src/snowcli/api/__init__.py

```diff
@@ -0,0 +1,32 @@
+from typing import Optional
+
+from snowcli.api.plugin.plugin_config import PluginConfigProvider
+
+
+class Api:
+    def __init__(
+        self,
+        plugin_config_provider: PluginConfigProvider,
+    ):
+        self.plugin_config_provider = plugin_config_provider
+
+
+class ApiNotInitializedError(RuntimeError):
```

Comment:
Maybe move it to `snowcli/exception.py`?

Reply:
`api` package shouldn't use classes/functions placed outside of it